modin/experimental/core/execution/native/implementations/hdk_on_native/dataframe/dataframe.py

```diff
@@ -419,15 +428,60 @@ def take_2d_labels_or_positional(
         if row_labels is not None:
             raise NotImplementedError("Row labels masking is not yet supported")
 
-        if row_positions is not None:
-            base = base._maybe_materialize_rowid()
-            op = MaskNode(base, row_labels=row_labels, row_positions=row_positions)
-            return self.__constructor__(
-                columns=base.columns,
-                dtypes=base.copy_dtypes_cache(),
-                op=op,
-                index_cols=base._index_cols,
-                force_execution_mode=base._force_execution_mode,
+        if row_positions is None:
+            return base
+
+        # Check if row_positions could be converted to a range.
+        if len(row_positions) > 2 and not is_range_like(row_positions):
```

Comment:
Is this related to the PR's goal?

Reply:
It's not. This PR changed execution of some operations, that were previously executed via arrow and now - via HDK. It caused the `asv_bench` failures. The failures occured due to a long range-like list of `row_positions`, that was converted to a huge json query, that HDK failed to execute.

Reply:
@AndreyPavlenko It's better to put it in a separate function.